### ISSUE TITLE
Incorrect python dataclass generation

### ISSUE THREAD

**Issue by melonora** [linkml/linkml#1352]
**Describe the bug**
Gen-python does not throw error when a slot has both range and any_of defined, while gen-pydantic does throw an error. Instead gen-python creates a dataclass where the type hint for the specific field just indicates the default range.

**To reproduce**

1. Create a yaml file and define a slot such as:
    ```
    any_of:
        - range: class1
        - range: class2
    ```
 2. run gen-python and gen-pydantic on the yaml file. Gen-pydantic will throw an error while gen-python creates a python script with a dataclass where the field is as described above.

**Expected behavior**
Generate a dataclass with a field having a typehint of both the ranges specified.

**Comment by cmungall** [comment]
Fixed by #1617

**Comment by melonora** [comment]
Awesome! I will close
